### PR TITLE
Added syslog configuration for OMERO server

### PR DESCRIPTION
**Pull request by dpwrussell**
Document use of syslog in logback for sysadmins.

### CODE REVIEW

**Comment by snoopycrimecop** [comment]
Empty PR description. Please add a short summary of the PR scope and some testing instructions.

**Comment by joshmoore** [comment]
NB: @dpwrussell don't know how useful it'd be for you, but I'd assume it'd be possible to use https://docs.python.org/2/library/logging.handlers.html#sysloghandler

**Comment by mtbc** [comment]
Staged at http://www.openmicroscopy.org/site/support/omero5.2-staging/sysadmins/server-syslog.html.

**Comment by dpwrussell** [comment]
@joshmoore For OMERO.web logging you mean?

**Comment by manics** [comment]
@dpwrussell I think logback.xml will only configure logging for the Java components, the Python ones (Processor, Tables) will still log to files.

File: omero/sysadmins/server-syslog.txt [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+    Finally, syslog can be configured to specifically output this facility
+    output to a file such as ``/var/log/omero``.
+
+Create the new file ``OMERO.server/current/etc/logback_syslog.xml``:
+
+::
+
+    <?xml version="1.0" encoding="UTF-8"?>
+    <included>
+        <!-- syslog -->
```
Comment (ximenesuk) [inline omero/sysadmins/server-syslog.txt]: This line is indented a couple of chars more than needed

File: omero/sysadmins/server-syslog.txt [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+        <facility>local6</facility>
+        <suffixPattern>OMERO [%level] [%thread] %logger %msg</suffixPattern>
+      </appender>
+
+    </included>
+
+This creates an appender that logs to syslog. `syslogHost` is the host on
+which syslog is running. No port is specified as `514` is the default. The
+`suffixPattern` is customizable. In this instance it is identical to OMERO's
+FILE logger except an added "OMERO" identifier has been added for clarity. The
```
Comment (ximenesuk) [inline omero/sysadmins/server-syslog.txt]: I don't think this use of FILE should be in uppercase as its not referring to the appender name.

File: omero/sysadmins/server-syslog.txt [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+        </filter>
+
+        <syslogHost>localhost</syslogHost>
+        <facility>local6</facility>
+        <suffixPattern>OMERO [%level] [%thread] %logger %msg</suffixPattern>
+      </appender>
+
+    </included>
+
+This creates an appender that logs to syslog. `syslogHost` is the host on
```
Comment (ximenesuk) [inline omero/sysadmins/server-syslog.txt]: I don't know whether all uses of 'syslog' need to be marked up or not. /cc @hflynn

Comment (hflynn) [inline omero/sysadmins/server-syslog.txt]: I wouldn't bother, don't think it would add to the readability.

**Comment by dpwrussell** [comment]
@manics Definitely it will only do logging for the Java processes, I was asking @joshmoore if he was suggesting modifying OMERO.web and other Python components to use the Python sysloghandler to add that as well.

There aren't obvious configuration files for that so I'm guessing it will be more complicated.

**Comment by dpwrussell** [comment]
All comments addressed.

**Comment by hflynn** [comment]
Is this ready to merge now or does it need relisting for someone else to look at it this week?

**Comment by ximenesuk** [comment]
Sorry, the changes are all good and it reads just fine. I have not checked the workflow myself as setting up syslog on my Mac involves messing around with plists. However, it all makes sense. I'm happy for it to be merged.

**Comment by hflynn** [comment]
@joshmoore do you want the workflow tested before we merge this?

**Comment by joshmoore** [comment]
Probably inclined to trust Douglas on this one. That being said, there's a longer term question of how to make sure this doesn't fall into disrepair over time. One way of doing that might be moving the logback_syslog.xml file into the server repo itself and potentially evening adding an environment based switch in the main logback.xml so that something like `OMERO_SYSLOG=true bin/omero admin start` would be enough to turn on the feature. This would then likely need to get tested (e.g. in omero-install).

Open to suggestions, but if need be, we can use a card as a follow-up. Thanks again for this, @dpwrussell!

**Comment by hflynn** [comment]
I'll merge and leave you to file a card in the appropriate place re: maintenance then @joshmoore

**Comment by joshmoore** [comment]
Filed: https://trello.com/c/r5TopDVf/671-syslog-memcached-testing

**Pull request merged**